src/TMap.cpp

```diff
@@ -1916,7 +2024,12 @@ bool TMap::retrieveMapFileStats(QString profile, QString* latestFileName = nullp
 
 int TMap::createMapLabel(int area, QString text, float x, float y, float z, QColor fg, QColor bg, bool showOnTop, bool noScaling, qreal zoom, int fontSize)
 {
-    if (!mpRoomDB->getArea(area)) {
+    auto pA = mpRoomDB->getArea(area);
+    if (!pA) {
+        return -1;
+    }
+
+    if (text.isEmpty()) {
```

Comment:
Can't labels have no text if they're an image label?

Reply:
I got to look more deeply into the code recently and it turns out that that field is used for the pathFileName of the file used as the source of the image - for an image label. However It turns out that there is NOT a specific indicator (e.g. `bool` isImageLabel) in the `TMapLabel` class to say which this is!

Reply:
Why do we need the path file name? Once the image is loaded, the external file is never referenced anymore?

Reply:
> Can't labels have no text if they're an image label? Map painting code allows for empty labels

They are created with a different method `createMapImageLabel(...)` !

Also, having the pathFileName to the image source will be desirable if someone ever gets around to improving the map label editing UI.

Reply:
Actually I got that wrong - the pathFileName is not stored - but it would make sense if it was - relative to the profile's base directory - so that an enhanced editor could go straight to it if asked...

Reply:
Adding new features should be seperate  from refactoring

Reply:
Probably - but this is just replicating the behaviour of the previous code (on old line 1933)...